openid-4-verifiable-presentations-1_0.md

```diff
@@ -559,11 +565,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array.
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The same basic logic applies for selecting claims and for selecting credentials, as detailed in the following.
+
+#### Selecting Claims
+
+The following rules apply for selecting claims via `expect_claims` and `valid_claim_sets`:
+
+- If `expect_claims` is not provided, the Verifier requests all claims existing
+  in the Credential.
+- If `expect_claims` is provided, but `valid_claim_sets` is not provided,
+  the Verifier requests all claims listed in `expect_claims`.
+- Otherwise, the Verifier requests one combination of the claims listed in
+  `valid_claim_sets`, with optional claims marked by the postfix `?`.
+
+If the Wallet cannot deliver all non-optional claims requested by the Verifier according to these rules, it MUST NOT
+return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `expect_credentials` and `valid_credential_sets`:
+
+- If `valid_credential_sets` is not provided, the Verifier requests all
+  Credentials in `expect_credentials` to be returned.
+- Otherwise, the Verifier requests one combination of the Credentials
+  listed in `valid_credential_sets`, with optional credentials marked by the postfix `?`.
+
+Credentials not matching the respective constraints expressed within
+`expect_credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the Verifier according to these rules, it MUST NOT
+return any credential.
+
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
+
+### Format `vc+sd-jwt` {#format_vc_sd_jwt}
+
+`vct_values`:
+: OPTIONAL. An array of strings that specifies allowed values for
+the type of the requested Verifiable Credential. All elements in the array MUST
+be valid type identifiers as defined in [@!I-D.ietf-oauth-sd-jwt-vc]. The Wallet
+MAY return credentials that inherit from any of the specified types, following
+the inheritance logic defined in [@!I-D.ietf-oauth-sd-jwt-vc].
+
+`credential_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies
+the allowed algorithms for signing the SD-JWT (i.e., only the issuer-signed part).
+
+`kbjwt_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies the
+allowed algorithms for signing the Key-Binding JWT (KB-JWT) (i.e., the part signed
+by the holder).
+
+`sd_alg_values`:
+: OPTIONAL. An array of strings that specifies the allowed hash algorithms for
+digests in the SD-JWT and KB-JWT.
```

Comment:
Prefer we remove these, as discussed previously I don't believe a usecase has been properly elaborated on and this creates needless complexity for majority of implementations and in worse cases possible security vulnerabilities through possible downgrade attacks.